cli/internal/fs/turbo_json.go

```diff
@@ -194,3 +206,40 @@ func (c *TaskDefinition) UnmarshalJSON(data []byte) error {
 	c.OutputMode = rawPipeline.OutputMode
 	return nil
 }
+
+// UnmarshalJSON deserializes TurboJSON objects into struct
+func (c *TurboJSON) UnmarshalJSON(data []byte) error {
+	raw := &rawTurboJSON{}
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	envVarDependencies := make(util.Set)
+	globalFileDependencies := make(util.Set)
+
+	for _, value := range raw.GlobalEnv {
+		if strings.HasPrefix(value, envPipelineDelimiter) {
+			return fmt.Errorf("You specified \"%s\" in the \"env\" key. This is treated as a literal, you may want to remove the \"%s\"", value, envPipelineDelimiter)
+		}
+
+		envVarDependencies.Add(value)
+	}
+
+	for _, value := range raw.GlobalDependencies {
+		if strings.HasPrefix(value, envPipelineDelimiter) {
+			envVarDependencies.Add(strings.TrimPrefix(value, envPipelineDelimiter))
+		} else {
+			globalFileDependencies.Add(value)
+		}
+	}
```

Comment:
All of this could be a shared utility function across both places.

Reply:
will defer this to some future time!